### PR TITLE
Add file attribute upload to product attributes

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds file attribute upload to product attributes.

~~:warning: This PR contains changes from https://github.com/mirumee/saleor-dashboard/pull/832 and https://github.com/mirumee/saleor-dashboard/pull/888, which should be merged first.~~
~~:warning: This PR contains changes from https://github.com/mirumee/saleor-dashboard/pull/888, which should be merged first.~~

Feature branch: https://github.com/mirumee/saleor-dashboard/pull/884

**PR intended to be tested with API branch:**  feature/file-uploads
https://github.com/mirumee/saleor/pull/6568

### Screenshots

In product create and product details pages:
Zrzut ekranu 2020-12-7 o 12 36 39

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://feature-file-uploads.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1343 Add suport for file selection and upload in attributes sections](https://app.clickup.com/t/2549495/SALEOR-1343)

File: src/products/views/ProductCreate/handlers.ts [hunk 1]
```diff
@@ -69,12 +82,39 @@ export function createHandler(
...
+          ...attributesWithAddedFiles,
+          {
+            file: uploadFileResult.data.fileUpload.uploadedFile.url,
+            id: attribute.id,
+            values: []
+          }
+        ];
+      },
+      []
+    );
```
Comment (dominik-zeglen) [inline src/products/views/ProductCreate/handlers.ts:110]: Can we move this somewhere else to increase readability?

File: src/products/views/ProductUpdate/handlers.ts [hunk 2]
```diff
@@ -110,16 +128,53 @@ export function createUpdateHandler(
...
 ) {
   return async (data: ProductUpdatePageSubmitData) => {
+    let errors: Array<
+      | ProductErrorFragment
+      | StockErrorFragment
+      | BulkStockErrorFragment
+      | AttributeErrorFragment
+      | UploadErrorFragment
+      | ProductChannelListingErrorFragment
+    > = [];
```
Comment (dominik-zeglen) [inline src/products/views/ProductUpdate/handlers.ts:144]: Let it be a separate type

File: src/products/views/ProductUpdate/handlers.ts [hunk 3]
```diff
@@ -110,16 +128,53 @@ export function createUpdateHandler(
...
+          ...attributesWithAddedFiles,
+          {
+            file: uploadFileResult.data.fileUpload.uploadedFile.url,
+            id: attribute.id,
+            values: []
+          }
+        ];
+      },
+      []
+    );
```
Comment (dominik-zeglen) [inline src/products/views/ProductUpdate/handlers.ts:169]: I believe I already saw it somewhere... 🤔

File: src/products/views/ProductUpdate/handlers.ts [hunk 4]
```diff
@@ -206,6 +254,34 @@ export function createUpdateHandler(
...
+        })
+      );
+
+      deleteAttributeValuesResult.forEach(deleteValueResult => {
+        errors = [
+          ...errors,
+          ...deleteValueResult.data.attributeValueDelete.errors
+        ];
+      });
+    }
```
Comment (dominik-zeglen) [inline src/products/views/ProductUpdate/handlers.ts:283]: Wouldn't `map` look better here?

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/products/views/ProductCreate/handlers.ts (hunk 1 above)
Comment (orzechdev) [reply src/products/views/ProductCreate/handlers.ts:110]: sure

File: src/products/views/ProductUpdate/handlers.ts (hunk 4 above)
Comment (orzechdev) [reply src/products/views/ProductUpdate/handlers.ts:283]: Not really, it merges one error array with many consecutive error arrays, so map can't help.

Comment (dominik-zeglen) [reply src/products/views/ProductUpdate/handlers.ts:276]: `reduce` then?

Comment (orzechdev) [reply src/products/views/ProductUpdate/handlers.ts]: might be, updated

**Review by mmarkusik — APPROVED** [review]

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**